### PR TITLE
New service and method API (SHOOP-2293)

### PR DESCRIPTION
**Pull request by suutari-ai**
Refactor payment and shipping methods from a BaseMethodModule based API
to a new polymorphic model based API with behavior components.

Most of these changes are pretty much unsplittable and therefore in a
single commit.  Sorry!

There is also some additions of utilities and some tests which are
separated from the main commit.

Payment and shipping method management from Shop Admin is currently
removed.  New admin modules will follow in other PRs.

### CODE REVIEW

**Comment by suutari-ai** [comment]
Just noticed that Change Log entries are missing. Please don't merge this just yet. Will add the change log entries soon...

File: shoop/core/models/_service_shipping.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+class CustomCarrier(Carrier):
+    """
+    Carrier without any integration or special processing.
+    """
+    class Meta:
+        verbose_name = _("custom carrier")
+        verbose_name_plural = _("custom carriers")
+
+    def get_service_choices(self):
+        return [ServiceChoice('manual', _("Manually processed shiment"))]
```
Comment (tulimaki) [inline shoop/core/models/_service_shipping.py]: shipment

Comment (suutari-ai) [inline shoop/core/models/_service_shipping.py]: Oops.. at least it's not shitment :) Will fix it. Thanks!

File: shoop/core/migrations/0020_services_and_methods.py [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+
+
+def create_default_service_providers(apps, schema_editor):
+    """
+    Create default carrier and payment processor.
+
+    Link existing ShippingMethods and PaymentMethods which use the
+    previous default module to the created carrier or payment processor.
+
+    Update polymorphic_ctypes for CustomerCarrier and
```
Comment (tulimaki) [inline shoop/core/migrations/0020_services_and_methods.py]: This method does not update polymorphic_ctypes anymore. Can you move the this comment.

**Comment by tulimaki** [comment]
This looks good. Mergeable after those comments above is solved. I think we should wait a little bit for the admin-side PRs.

**Comment by suutari-ai** [comment]
* Fixed the "shiment" typo.
* Added change log entries.
* Fixed docstrings in migrations.
* Removed admin form provides from shoop.testing, since they don't do anything yet.

**Comment by tulimaki** [comment]
Great, let's do this!

**Pull request merged**